File: 백준/Bronze/2846. 오르막길/오르막길.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

class Main{
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static StringTokenizer st;
    static StringBuilder sb = new StringBuilder();

    public static void main(String[]args) throws IOException {
        int N = Integer.parseInt(br.readLine().trim());
        int[] h = new int[N];

        st = new StringTokenizer(br.readLine().trim());
        for(int i=0; i<N; i++){
            h[i] = Integer.parseInt(st.nextToken());
        }

        int max = 0;
        int start = h[0];

        for (int i = 1; i < N; i++) {
            if (h[i] > h[i - 1]) {
                // 오르막길
                max = Math.max(max, h[i] - start);
            } else {
                // 내리막, 평지
                start = h[i];
            }
        }
        sb.append(max);

        System.out.print(sb);
    }
}

File: 백준/Bronze/2846. 오르막길/오르막길.py
import sys
input = sys.stdin.readline

N = int(input())
h = list(map(int, input().split()))

start = h[0]
maxHeight = 0;

for i in range(1, N):    
    if(h[i-1]<h[i]): # 오르막길일때
        maxHeight = max(maxHeight, h[i]-start)
    else: # 내리막길일때
        start = h[i]        

print(maxHeight)